### PR TITLE
check sparse sizes

### PR DESCRIPTION
**Pull request by ngimel**
checks sizes of sparse tensors when comparing them in assertEqual.
Removes additional checks in safeCoalesce, safeCoalesce should not be a test for `.coalesce()` function.

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -1080,6 +1047,13 @@ def assertEqual(self, x, y, msg: Optional[str] = None, *,
             super().assertEqual(x.is_sparse, y.is_sparse, msg=msg)
             super().assertEqual(x.is_quantized, y.is_quantized, msg=msg)
             if x.is_sparse:
+                if x.size() != y.size():
+                    debug_msg_sparse = ("Attempted to compare equality of tensors with the different sizes. "
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:1051]: "with the different sizes." -> "with different sizes!"

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -883,40 +883,7 @@ def safeCoalesce(self, t):
...
-
-        self.assertEqual(tc._indices(), tg._indices())
-        self.assertEqual(tc._values(), tg._values())
-
-        if t.is_coalesced():
-            self.assertEqual(tc._indices(), t._indices())
-            self.assertEqual(tc._values(), t._values())
-
-        return tg
+        return tc
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:886]: What's going on here?

I'm not sure why this function or its sister function `safeToDense` even exist. Should this PR remove them? 

I can see one case where it's logic may need to be preserved:

https://github.com/pytorch/pytorch/blob/1cc1da541158b53c24a579fd316d190d5c27bfd1/test/test_sparse.py#L200

Comment (ngimel) [reply torch/testing/_internal/common_utils.py:886]: This function moonlights as a correctness test for `.coalesce`, and thus should be removed. `test_coalesce` should do its own correctness testing. I pretty much removed all of this function, I can fully remove it and move the logic to test_coalesce.

Comment (mruberry) [reply torch/testing/_internal/common_utils.py:886]: That sounds good

**Review by mruberry — APPROVED** [review]
Cool!

**Pull request closed without merging**